python/ml4ir/base/features/preprocessing.py

```diff
@@ -53,6 +58,71 @@ def preprocess_text(feature_tensor, remove_punctuation=False, to_lower=False):
     return feature_tensor
 
 
+def get_one_hot_label_vectorizer(feature_info, file_io: FileIO):
+    """
+    Returns a tf function to convert categorical string labels to a one hot encoding.
+
+    Args:
+        feature_info: Dictionary representing the configuration parameters for the specific feature from the FeatureConfig.
+                      See categorical_indicator_with_vocabulary_file, here it is used to read a vocabulary file to
+                      create the one hot encoding.
+        file_io: FileIO required to load the vocabulary file.
+
+    Returns:
+        processed float tensor
+    """
+    # Avoid initialization if there is no vocabulary file stated in feature config wrt to label definition.
+    if not ("args" in feature_info.get("feature_layer_info")) or not (
```

Comment:
@Ullimague Why are you using `feature_layer_info` here? Try using `preprocessing_info` and pass args using that.

Reply:
Otherwise you will also be accidentally running this in the feature layer step.

Reply:
Because if there is no vocabulary file stated in `feature_layer_info`, we won't be able to initialize the `get_one_hot_label_vectorizer` to `PreprocessingMap`. The vocabulary file was passed via the `feature_layer_info`, and not the `preprocessing_info`.

Reply:
For instance a config could have a preprocessing_info, but no feature_layer_info. I imagine we still want to authorize this case? @lastmansleeping